### PR TITLE
[9.0][ADD]purchase request to rfq

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
# Purchase Request to RFQ

This module adds the possibility to create or update Requests for Quotation (RFQ) from Purchase Request Lines. This module depends on purchase request, which is used to give notification of requirements of materials . The PR for this one is here: https://github.com/OCA/purchase-workflow/pull/259
## Usage
- Select the lines that you wish to initiate the RFQ for, then go to 'More' and press 'Create RFQ'.
- You can choose to select an existing RFQ or create a new one. In the later, you have to choose a supplier.
- In case that you chose to select an existing RFQ, the application will search for existing lines matching the request line, and will add the extra quantity to them

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
cc @jbeficent

File: purchase_request_to_rfq/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+{
+    "name": "Purchase Request to RFQ",
+    "author": "Eficent,"
+              "Odoo Community Association (OCA)",
+    "version": "8.0.1.0.0",
```
Comment (adrienpeiffer) [inline purchase_request_to_rfq/__openerp__.py]: `9.0.1.0.0` I'll correct this one in a PR to add setup file

File: purchase_request_to_rfq/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+{
+    "name": "Purchase Request to RFQ",
+    "author": "Eficent,"
+              "Odoo Community Association (OCA)",
+    "version": "9.0.1.0.0",
+    "contributors": [
```
Comment (adrienpeiffer) [inline purchase_request_to_rfq/__openerp__.py]: Can you remove this section from the manifest ?

**Comment by adrienpeiffer** [comment]
I think it shouldn't be possible to create RFQ from unapproved purchase request line. Currenly, it's possible ....
@aheficent @jbeficent What do you think ?

File: purchase_request_to_rfq/models/purchase_request.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    def _calc_new_qty_price(self, request_line, po_line=None, cancel=False):
+        uom_obj = self.env['product.uom']
+        qty = uom_obj._compute_qty(request_line.product_uom_id.id,
+                                   request_line.product_qty,
+                                   request_line.product_id.uom_po_id.id)
+        # Make sure we use the minimum quantity of the partner corresponding
+        # to the PO. This does not apply in case of dropshipping
+        supplierinfo_min_qty = 0.0
+        if po_line.order_id.location_id.usage != 'customer':
+            if po_line.product_id.seller_id.id == \
```
Comment (adrienpeiffer) [inline purchase_request_to_rfq/models/purchase_request.py]: Is it possible that `seller_id` doesn't exist in version 9 ?

File: purchase_request_to_rfq/models/purchase_request.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                    po_line.order_id.partner_id.id:
+                supplierinfo_min_qty = po_line.product_id.seller_qty
```
Comment (adrienpeiffer) [inline purchase_request_to_rfq/models/purchase_request.py]: Same question about `seller_qty`

File: purchase_request_to_rfq/models/purchase_request.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    @api.model
+    def _calc_new_qty_price(self, request_line, po_line=None, cancel=False):
+        uom_obj = self.env['product.uom']
+        qty = uom_obj._compute_qty(request_line.product_uom_id.id,
+                                   request_line.product_qty,
+                                   request_line.product_id.uom_po_id.id)
+        # Make sure we use the minimum quantity of the partner corresponding
+        # to the PO. This does not apply in case of dropshipping
+        supplierinfo_min_qty = 0.0
+        if po_line.order_id.location_id.usage != 'customer':
```
Comment (adrienpeiffer) [inline purchase_request_to_rfq/models/purchase_request.py]: Is it possible that `location_id` doesn't exist in version 9 ?

File: purchase_request_to_rfq/models/purchase_request.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+                qty += uom_obj._compute_qty(rl.product_uom_id.id,
+                                            rl.product_qty,
+                                            rl.product_id.uom_po_id.id)
+            qty = max(qty, supplierinfo_min_qty) if qty > 0.0 else 0.0
+
+        price = po_line.price_unit
+        if qty != po_line.product_qty:
+            pricelist_obj = self.pool['product.pricelist']
+            pricelist_id = po_line.order_id.partner_id.\
+                property_product_pricelist_purchase.id
```
Comment (adrienpeiffer) [inline purchase_request_to_rfq/models/purchase_request.py]: same question about `property_product_pricelist_purchase`

Comment (AaronHForgeFlow) [reply purchase_request_to_rfq/models/purchase_request.py]: You're right @adrienpeiffer  Thanks a lot!

**Comment by adrienpeiffer** [comment]
@aheficent There is some changes in other module ...

**Comment by AaronHForgeFlow** [comment]
I'm sorry @adrienpeiffer  I did not understand. Do you mean I need to rebase this branch?

**Comment by adrienpeiffer** [comment]
No, if you look on files changed for this PR, some changes appear for other module probably due to merge or rebase.

**Comment by AaronHForgeFlow** [comment]
Oh, what a mess, thank you!

**Pull request closed without merging**